### PR TITLE
Convert to new override syntax

### PR DESCRIPTION
**Pull request by shr-project**

### CODE REVIEW

**Comment by srwalter** [comment]
Need to update YOCTO_BRANCH in .github/workflows/main.yml to fix CI

**Comment by cmcqueen** [comment]
What minimum version of Yocto is needed for this new override syntax? Please continue to support Yocto dunfell (which is LTS) if at all possible.

**Comment by srwalter** [comment]
I would also like to see us maintain dunfell support.  Does honister drop support for the older override syntax?  If not I'd be inclined to stay with the old syntax until an LTS comes out that supports it.

**Comment by shr-project** [comment]
dunfell, gatesgarth, hardknott and honister support this syntax, you just need to use latest revision from corresponding bitbake branch (1.46, 1.48, 1.50, master). See https://git.openembedded.org/bitbake/commit/?h=1.46&id=c5418eae56cc50dbae7951c055434a0c456c53a4 for dunfell builds.

**Comment by shr-project** [comment]
But you're right that I should update 3rd commit to include dunfell, gatesgarth, hardknott now when bitbake changes were merged for them.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -2,8 +2,8 @@ on: [pull_request]
 jobs:
   build:
     env:
-      YOCTO_VERSION: 3.2.3
-      YOCTO_BRANCH: gatesgarth
+      YOCTO_VERSION: 3.3.2
+      YOCTO_BRANCH: hardknott
```
Comment (srwalter) [inline .github/workflows/main.yml:6]: As we discussed, I think we want meta-rust to support dunfell for the foreseeable future.  Therefore we should probably configure the CI to use dunfell

Comment (shr-project) [reply .github/workflows/main.yml:6]: Then there should be 2 or 3 CI jobs to test against dunfell, hardknott and honister (which is what this layer claims compatibility with) only the currently used gatesgarth is the least important one as gatesgarth is EOL.

Comment (srwalter) [reply .github/workflows/main.yml:6]: I don't disagree with that, but dunfell is what I want to see pass before merging this PR :smile:

Comment (shr-project) [reply .github/workflows/main.yml:6]: hardknott with 1.50 bitbake is affected the same as dunfell with 1.46 and both got the same patch in bitbake, so if hardknott is passing, then dunfell will as well (if it wasn't already failing before this PR).

Comment (otavio) [reply .github/workflows/main.yml:6]: I tend to agree with @srwalter here, and I'd support `dunfell` to be included in the test matrix. Besides `dunfell`, `master` would be great too. The other releases we can assume to be good if those two are good.

**Review by kraj — APPROVED** [review]
lgtm

File: .github/workflows/main.yml (hunk 1 above)
Comment (shr-project) [reply .github/workflows/main.yml:6]: I've triggered CI for dunfell, hardknott and master with this PR and all 3 were canceled after 5 hours during testing SDK tasks, but you can check the logs that it parsed and built something.

Comment (shr-project) [reply .github/workflows/main.yml:6]: And the last run with hardknott again failed due to disk space:
```
WARNING: The free space of /home/srwalter/actions-runner/_work/meta-rust/meta-rust/poky/build/tmp (/dev/mapper/ubuntu--vg-ubuntu--lv) is running low (0.097GB left)
WARNING: The free space of /home/srwalter/actions-runner/_work/meta-rust/meta-rust/poky/build/downloads (/dev/mapper/ubuntu--vg-ubuntu--lv) is running low (0.097GB left)
WARNING: The free space of /home/srwalter/actions-runner/_work/meta-rust/meta-rust/poky/build/sstate-cache (/dev/mapper/ubuntu--vg-ubuntu--lv) is running low (0.097GB left)
```

**Comment by aboseley** [comment]
What is the advantage of upgrading to the new syntax?  This change reduces the availability of this meta layer to older poky releases?

**Comment by shr-project** [comment]
> What is the advantage of upgrading to the new syntax? This change reduces the availability of this meta layer to older poky releases?

It's needed for honister and compatible with dunfell, gatesgarth, hardknott releases as well. Zeus and older releases aren't supported for long time, if you need to use meta-rust with zeus or older, use older meta-rust revision or ask for zeus branch to be created.

**Comment by mlilien** [comment]
Hi, for me it is not working with the latest poky dunfell release 3.1.10.
I get an error 
```--- Starting bitbake server pid 184 at 2021-08-18 09:21:04.326027 ---
ERROR: ParseError at /yocto-build/yocto-source/meta-rust/conf/distro/include/rust_security_flags.inc:2: unparsed line: 'SECURITY_CFLAGS:pn-rust-native = "${SECURITY_NO_PIE_CFLAGS}"'````

**Comment by shr-project** [comment]
> Hi, for me it is not working with the latest poky dunfell release 3.1.10.
> I get an error
> 
> ```
> ERROR: ParseError at /yocto-build/yocto-source/meta-rust/conf/distro/include/rust_security_flags.inc:2: unparsed line: 'SECURITY_CFLAGS:pn-rust-native = "${SECURITY_NO_PIE_CFLAGS}"'````
> ```

Use latest revision from dunfell branch (not 3.1.10 point release).

**Comment by mlilien** [comment]
Thank you, for clarifying that.

**Pull request merged**